### PR TITLE
Regular polygons

### PR DESCRIPTION
**Pull request by joanita-51**
Hello! I've successfully implemented regular polygons as outlined in issue #26. I've added support for drawing square, pentagon, and hexagon using the draw_poly function.

Here are some screenshots showcasing the new functionality:

### Square Polygon:
Screenshot from 2023-10-09 12-47-56

### Pentagon Polygon:
image

### Hexagon Polygon:
image

Please review and merge this pull request at your convenience. If you have any feedback or questions, feel free to let me know. Thanks!

### CODE REVIEW

**Comment by nikochiko** [comment]
@joanita-51 Great work! 🎉 

> Make regular polygons -- triangle, square, pentagon, hexagon -- in the same canvas.

Can you please put all shapes in the same file? It's up to you how you want to align them -- concentric, as a grid of 4, aligned in a row or a column -- however you like it.

Don't worry about making types and functions with center points to do this. It's okay to just use hardcoded numbers too for points. The point of this exercise is to become familiar with the API for graphics library, so we can look at these examples when adding something new later.

File: examples/pentagon.ml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      let x = pent.c.x in
+      let y = pent.c.y in
+      let points =
+        [|
+          (x, y + side);
+          (x - side, y);
+          (x - (side/2), y - side);
+          (x + (side/2), y - side);
+          (x + side, y);
+        |]
```
Comment (nikochiko) [inline examples/pentagon.ml:24]: This does not create a regular pentagon. Some sides are longer than others.

e.g. you can use a calculator to get the 5 coordinates of a regular pentagon for some constant k with this calculation:
```
1. (0, k)
2. (-k * cos18, k * sin18)
3. (-k * cos54, -k * sin54)
4. (k * cos54, -k * sin54)
5. (k * cos18, k * sin18) 
```

(54 and 18 are in degrees)

Comment (joanita-51) [reply examples/pentagon.ml:24]: I'm encountering the following when i try to run the file 
image

Here is the modified code
```
open Graphics

type point = { x : int; y : int }
type pentagon = {c : point; side_length : int}

type shape = Pentagon of pentagon

let canvas_size = (500, 500)
let canvas_mid = { x = fst canvas_size / 2; y = snd canvas_size / 2 }

let render_shape s =
  match s with
  | Pentagon pent ->
    let k = pent.side_length in
    let points =
      [|
        (0, k);
        (int_of_float (-. (float_of_int k) *. cos (18. *. (Float.pi /. 180.))), int_of_float ((float_of_int k) *. sin (18. *. (Float.pi /. 180.))));
        (int_of_float (-. (float_of_int k) *. cos (54. *. (Float.pi /. 180.))), int_of_float (-. (float_of_int k) *. sin (54. *. (Float.pi /. 180.))));
        (int_of_float ((float_of_int k) *. cos (54. *. (Float.pi /. 180.))), int_of_float (-. (float_of_int k) *. sin (54. *. (Float.pi /. 180.))));
        (int_of_float ((float_of_int k) *. cos (18. *. (Float.pi /. 180.))), int_of_float ((float_of_int k) *. sin (18. *. (Float.pi /. 180.))));
      |]
    in
    draw_poly points


let pentagon ?x ?y side_length =
  let center = match (x, y) with
    | Some x, Some y -> { x; y }
    | _ -> canvas_mid in
  Pentagon { c = center; side_length }

let show shapes = List.iter render_shape shapes

let () =
  open_graph (" " ^ string_of_int (fst canvas_size) ^ "x" ^ string_of_int (snd canvas_size));
  set_color black;

  let pent = pentagon 100 in
  show [pent];

  ignore (read_line ());
  close_graph ()
```

Comment (nikochiko) [reply examples/pentagon.ml:24]: @joanita-51 this is too complicated for an example. I didn't want you to add all the calculations inside the file, but rather do one calculation and build an example with that.

Can you please do that? I calculated a set of 5 points with k=200, please use these in `draw_poly` to build your pentagon (on a 500x500 canvas) --

```
  (250, 450)
  (60, 312)
  (132, 88)
  (368, 88)
  (440, 312)
```

**Comment by joanita-51** [comment]
@nikochiko When I use the calculated values as my points, I receive the following warnings.
image
image

**Comment by nikochiko** [comment]
@joanita-51 the warnings are there because there are a bunch of unused fields. You can go ahead and remove that code. It isn't needed, you can directly pass the points to `Graphics.draw_poly`.

**Comment by joanita-51** [comment]
Thank you @nikochiko I've added the polygon points and also placed the polygons in the same file. To avoid overlap with other shapes, I halved the pentagon's size. Are there any other modifications I can make?
image

**Comment by nikochiko** [comment]
@joanita-51 thanks for the changes! this looks good to me.

can you add a screenshot of the rendered image now?

**Comment by joanita-51** [comment]
Thank you very much for your assistance, @nikochiko. Is this the image you are referring to as 'rendered'?
image

**Comment by nikochiko** [comment]
Yes! This is good enough. Thanks for the changes.

**Review by nikochiko — APPROVED** [review]

**Pull request merged**